test/teiserver/battle/balance_lib_internal_test.exs

```diff
@@ -0,0 +1,59 @@
+defmodule Teiserver.Battle.BalanceLibInternalTest do
+  @moduledoc """
+  Can run all balance tests via
+  mix test --only balance_test
+  """
+  use ExUnit.Case
+  import Mock
+  @moduletag :balance_test
+  alias Teiserver.Battle.BalanceLib
+
+  setup_with_mocks([
+    {Teiserver.Account, [:passthrough],
+     [get_user_by_id: fn member_id -> get_user_by_id_mock(member_id) end]}
+  ]) do
+    :ok
+  end
+
+  test "Able to standardise groups with incomplete data" do
+    groups = [
+      %{1 => 19, 2 => 20},
+      %{3 => 18},
+      %{4 => 15},
+      %{5 => 11}
+    ]
```

Comment:
I am not a fan at all of the use of mocks. It forces you to have a deep knowledge of the internals of the stuff you're testing. You recognise that with the comment a bit bellow:
> loser_picks algo will hit the databases so let's just test with split_one_chevs

It's also a departure from the other tests elsewhere.

For these reason, I'd rather have something that creates the mock data in the DB, and then exercise the code under test.

Using `DataCase, async: true`, I propose the following:
```elixir
    [user1, user2, user3, user4, user5] =
      Enum.map(1..5, fn k ->
        Teiserver.TeiserverTestLib.new_user("User_#{k}")
      end)

    groups = [
      %{user1.id => 19, user2.id => 20},
      %{user3.id => 18},
      %{user4.id => 15},
      %{user5.id => 11}
    ]

    fixed_groups = BalanceLib.standardise_groups(groups)

    assert fixed_groups == [
             %{
               user1.id => %{name: "User_1", rank: 0, rating: 19},
               user2.id => %{name: "User_2", rank: 0, rating: 20}
             },
             %{user3.id => %{name: "User_3", rank: 0, rating: 18}},
             %{user4.id => %{name: "User_4", rank: 0, rating: 15}},
             %{user5.id => %{name: "User_5", rank: 0, rating: 11}}
           ]
```

and similar for the other test. You could even consider pulling the creation of users in a setup function for the module.

Let me know if you have any question over this code.

Reply:
After creating these users, do they get deleted at some point at end of test?